brcm2708: Add feature flag rootfs-part

danielfdickinson opened the pull request:
Even with squashfs brcm2708 requires ROOTFS_PART_SIZE because the overlay
exists as a loopback device on the space not used by squashfs in the root
partition.  Also for ext4 (the other fs option) ROOTFS_PART_SIZE is required,
so always enable ROOTFS_PART_SIZE for brcm2708.

Signed-off-by: Daniel F. Dickinson <cshored@thecshore.com>

cshoredaniel commented:
Ooops, forgot to mention the following errors appear if this isn't done:

```
+ '[' 4 -eq 5 ']'
+ echo 'SYNTAX: ./gen_rpi_sdcard_img.sh <file> <bootfs image> <rootfs image> <bootfs size> <rootfs size>'
SYNTAX: ./gen_rpi_sdcard_img.sh <file> <bootfs image> <rootfs image> <bootfs size> <rootfs size>
+ exit 1
make[5]: *** [Makefile:103: /home/daniel/Build/openwrt-pi/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/linux-brcm2708_bcm2708/tmp/openwrt-dvc-4-snapshot-r8858-b53111ad42-brcm2708-bcm2708-rpi-squashfs-factory.img.gz] Error 1
make[5]: *** Waiting for unfinished jobs....
+ '[' 4 -eq 5 ']'
+ echo 'SYNTAX: ./gen_rpi_sdcard_img.sh <file> <bootfs image> <rootfs image> <bootfs size> <rootfs size>'
SYNTAX: ./gen_rpi_sdcard_img.sh <file> <bootfs image> <rootfs image> <bootfs size> <rootfs size>
+ exit 1
make[5]: *** [Makefile:103: /home/daniel/Build/openwrt-pi/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/linux-brcm2708_bcm2708/tmp/openwrt-dvc-4-snapshot-r8858-b53111ad42-brcm2708-bcm2708-rpi-squashfs-sysupgrade.img.gz] Error 1

```

chunkeey commented:
Hmm, I think you could instead add the rootfs-part feature flag ( added in 00f030a9c67c551e675364d73234a4cba273b420 ) to brcm2708's target makefile.

cshoredaniel commented:
Working on it...will push once tested

Commit:
brcm2708: Add feature flag rootfs-part

Even with squashfs brcm2708 requires ROOTFS_PART_SIZE because the overlay
exists as a loopback device on the space not used by squashfs in the root
partition. Also for ext4 (the other fs option) ROOTFS_PART_SIZE is required,
so use feature flag rootfs-part to enable it.

Signed-off-by: Daniel F. Dickinson <cshored@thecshore.com>

chunkeey commented:
Merged. Thanks.
